### PR TITLE
added VALARM to birthday events

### PR DESCRIPTION
**Pull request by Garfonso**
I did this small change for my owncloud installation, because I want to get a reminder on my mobile for contact birthdays.

Some devices might generate audible ALARMs at 0 o'clock from them, despite the VCALENDAR entry saying it should only display something. So it might be a good idea to make this optional or the offset configurable.. for example changing the trigger to PT12H would display the alarm at noon. 
Currently I can't see any options for the contacts app, so maybe this is not feasible either. 

Anyway, for me this works fine and I wanted to offer this change to you.

### CODE REVIEW

**Comment by jbtbnl** [comment]
This is great!

@tanghus what do you think, make the alarm time configurable or not?

**Comment by tanghus** [comment]
@Garfonso Thanks for your contribution. You're right that there should be a VALARM, but the current implementation will be changed in the near future because the two apps - contacts and calendar - are too tightly coupled.
We have a basic address book manager `OCP\Contacts` in core, that should be used by the calendar app instead. It needs to be improved first though as it doesn't support all that's needed for calendar app to build the birthday events.
I don't just want to add helper methods to that class, but prefer to make a couple of interfaces that can be used in more contexts than just getting address books and contexts.
So I'm not really that happy with adding features to a method that will be removed soon, but will leave this PR open so people can add the patch themselves until the proper implementation is ready.

**Comment by jancborchardt** [comment]
Hm – but wouldn’t having this by default be very annoying? Especially audible alarms should not be used lightly.

File: lib/contact.php [hunk 1]
```diff
@@ -766,6 +766,11 @@ public function getBirthdayEvent() {
 			$vevent->{'UID'} = $this->UID;
 			$vevent->{'RRULE'} = 'FREQ=YEARLY';
 			$vevent->{'SUMMARY'} = $title . ' (' . $date->format('Y') . ')';
+			$alarm = \Sabre\Vobject\Component::create('VALARM');
+			$alarm->{'TRIGGER'} = '-PT0M';
+			$alarm->{'ACTION'} = 'DISPLAY';
```
Comment (DeepDiver1975) [inline lib/contact.php]: @jancborchardt this specifies that the alarm should be displayed - there is no audio involved - see https://tools.ietf.org/html/rfc5545#section-3.6.6

**Comment by DeepDiver1975** [comment]
Sounds reasonable - please rebase @Garfonso THX

**Comment by Garfonso** [comment]
I rebased it. Not sure, if I messed something up or not... somehow some additional commits did show up. If that's an issue, you can close this and I redo the PR from a fresh clone.

**Comment by DeepDiver1975** [comment]
The commit history is a bit of a mess .. I'll fix this for you @Garfonso

**Comment by ghost** [comment]
I think this PR is still a bit of mess itself as it changes appinfo/app.php and  templates/admin.php which is not needed for this feature. Because of this there are also conflicts with the PR.

**Comment by Garfonso** [comment]
Yes, I somehow messed this up in a rebase. I don't know how and have no clue how to fix, tbh. So without assistance I'm not able clean up. :-(

**Comment by ghost** [comment]
Maybe the easiest thing is to create a new PR with only the changes within the lib/contact.php?

Btw. currently testing this and reminders are showing up fine in Thunderbird. However when "Dismissing" a reminder the following is shown within the data/owncloud.log on a oC 8.2.1 with contacts 0.5.0.0:

```
{"reqId":"lw5I5Za0gO01zqlsylBs","remoteAddr":"88.66.232.179","app":"PHP","message":"feof() expects parameter 1 to be resource, null given at \/var\/www\/owncloud\/3rdparty\/sabre\/vobject\/lib\/Parser\/MimeDir.php#245","level":3,"time":"2015-12-05T10:38:18+01:00"}
{"reqId":"lw5I5Za0gO01zqlsylBs","remoteAddr":"88.66.232.179","app":"PHP","message":"fgets() expects parameter 1 to be resource, null given at \/var\/www\/owncloud\/3rdparty\/sabre\/vobject\/lib\/Parser\/MimeDir.php#247","level":3,"time":"2015-12-05T10:38:18+01:00"}
{"reqId":"lw5I5Za0gO01zqlsylBs","remoteAddr":"88.66.232.179","app":"PHP","message":"feof() expects parameter 1 to be resource, null given at \/var\/www\/owncloud\/3rdparty\/sabre\/vobject\/lib\/Parser\/MimeDir.php#249","level":3,"time":"2015-12-05T10:38:18+01:00"}
{"reqId":"lw5I5Za0gO01zqlsylBs","remoteAddr":"88.66.232.179","app":"caldav","message":"Exception: {\"Message\":\"Error reading from input stream\",\"Exception\":\"Sabre\\\\VObject\\\\ParseException\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/vobject\\\/lib\\\/Parser\\\/MimeDir.php(102): Sabre\\\\VObject\\\\Parser\\\\MimeDir->readLine()\\n#1 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/vobject\\\/lib\\\/Parser\\\/MimeDir.php(63): Sabre\\\\VObject\\\\Parser\\\\MimeDir->parseDocument()\\n#2 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/vobject\\\/lib\\\/Reader.php(44): Sabre\\\\VObject\\\\Parser\\\\MimeDir->parse(NULL, 0)\\n#3 \\\/var\\\/www\\\/owncloud\\\/apps\\\/calendar\\\/lib\\\/object.php(249): Sabre\\\\VObject\\\\Reader::read(NULL)\\n#4 \\\/var\\\/www\\\/owncloud\\\/apps\\\/calendar\\\/lib\\\/sabre\\\/backend.php(410): OC_Calendar_Object::editFromDAVData('contact_birthda...', 'local::7::124.i...', 'BEGIN:VCALENDAR...')\\n#5 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/CalDAV\\\/CalendarObject.php(104): OCA\\\\Calendar\\\\Sabre\\\\Backend->updateCalendarObject('contact_birthda...', 'local::7::124.i...', 'BEGIN:VCALENDAR...')\\n#6 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1087): Sabre\\\\CalDAV\\\\CalendarObject->put('BEGIN:VCALENDAR...')\\n#7 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(501): Sabre\\\\DAV\\\\Server->updateFile('calendars\\\/Ranco...', Resource id #41, NULL)\\n#8 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#9 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#10 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(469): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#11 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#12 \\\/var\\\/www\\\/owncloud\\\/apps\\\/calendar\\\/appinfo\\\/remote.php(60): Sabre\\\\DAV\\\\Server->exec()\\n#13 \\\/var\\\/www\\\/owncloud\\\/remote.php(137): require_once('\\\/var\\\/www\\\/ownclo...')\\n#14 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/vobject\\\/lib\\\/Parser\\\/MimeDir.php\",\"Line\":253}","level":4,"time":"2015-12-05T10:38:18+01:00"}
```

Maybe related to the read-only calendar contacts_birthday provided by the contacts/calendar app?

**Comment by ghost** [comment]
> Btw. currently testing this and reminders are showing up fine in Thunderbird. However when "Dismissing" a reminder the following is shown within the data/owncloud.log on a oC 8.2.1 with contacts 0.5.0.0

I have this issue too after updating Contacts to 0.5.0.0. I will issue a new issue :smile:

**Comment by DeepDiver1975** [comment]
I'm closing this - starting with ownCloud 9.0 CalDAV and CardDAV are maintained within the core repo.

**Pull request closed without merging**